File: chapter 6/coffee_pos.pl
use strict;
use warnings;

use lib 'lib';
use Transactions ':all';
use Promotion;
use Starbuzz;

my @items = ("DONUT", "LATTE", "FILTER", "MUFFIN");
my @prices = (1.50, 2.0, 1.80, 1.20);

my $running = 1;
while ($running) {
    my $option = 1;
    for my $choice (@items) {
        print "$option $choice\n";
        $option++;
    }
    print "$option . Quit\n";
    print "Choose an option: ";
    my $choice = <STDIN>;
    if ($option == $choice) {
        $running = 0;
    } else {
        print "Credit card number: ";
        chomp(my $credit_card = <STDIN>);
        my $price = Promotion::discount($prices[$choice - 1]);
        print "Starbuzz card? ";
        chomp(my $answer = <STDIN>);
        if ($answer eq "Y") {
            $price = Starbuzz::discount($price);
        }
        save_transactions($price, $credit_card, $items[$choice - 1]);
    }
}

File: chapter 6/coffee_pos.py
def save_transactions(price, credit_card, description):
    file = open("transactions.txt", "a")
    file.write("%16d%07d%16s\n" % (credit_card, price * 100, description))
    file.close()

items = ["DONUT", "LATTE", "FILTER", "MUFFIN"]
prices = [1.50, 2.0, 1.80, 1.20]
running = True

while running:
    option = 1
    for choice in items:
        print(str(option) + " " + choice)
        option += 1
    print(str(option) + ". Quit")
    choice = int(input("Choose an option: "))
    if choice == option:
        running = False
    else:
        credit_card = int(input("Credit card number: "))
        save_transactions(prices[choice - 1], credit_card, items[choice -1])